### PR TITLE
child diff simplify check - compare delete and content first

### PR DESCRIPTION
**Pull request by nopcoder**

### CODE REVIEW

File: catalog/cataloger_diff.go [hunk 1]
```diff
@@ -414,48 +420,35 @@ func createDiffResultsTable(ctx context.Context, executor sq.Execer) (string, er
...
-			return DifferenceTypeRemoved
-		}
+
+	// same entry - none
+	if matchedParent != nil && childEnt.IsDeleted() == matchedParent.IsDeleted() && childEnt.Checksum == matchedParent.Checksum {
 		return DifferenceTypeNone
 	}
-	// when the entry was not deleted
+
+	// at this point both are not the same and not both are not deleted
```
Comment (tzahij) [inline catalog/cataloger_diff.go:434]: hard to understand comment - to many not

**Review by tzahij — APPROVED** [review]
Looks great

File: catalog/cataloger_diff.go [hunk 2]
```diff
@@ -366,7 +366,13 @@ func (c *cataloger) diffFromChild(ctx context.Context, tx db.Tx, childID, parent
...
 		}
 
-		diffType := evaluateFromChildElementDiffType(effectiveCommits, parentID, childEnt, parentEnt)
+		// point to matched child based on path
+		var matchedParent *DBScannerEntry
+		if parentEnt != nil && parentEnt.Path == childEnt.Path {
+			matchedParent = parentEnt
+		}
+
+		diffType := evaluateFromChildElementDiffType(effectiveCommits, childEnt, matchedParent)
```
[the remark this answers is not in the document]
Comment (nopcoder) [reply catalog/cataloger_diff.go:407]: the check is in place the previous check continue to fall if it is not a conflict

File: catalog/cataloger_diff.go (hunk 1 above)
Comment (nopcoder) [reply catalog/cataloger_diff.go]: removed one not

**Pull request merged**